frontend/src/common/backend.ts

```diff
@@ -163,6 +163,52 @@ export type VersionDto = {
   keycloakVersion: string;
 }
 
+enum AuditEventType {
+  Unlock = 'UNLOCK'
+}
+
+export abstract class AuditEventDto {
+  constructor(public id: number, public timestamp: Date, public type: AuditEventType) { }
+
+  abstract details(): { [key: string]: string };
+}
+
+enum UnlockResult {
+  Success = 'SUCCESS',
+  NoSuchDevice = 'NO_SUCH_DEVICE',
+  DeviceNotAuthorized = 'DEVICE_NOT_AUTHORIZED'
+}
```

Comment:
Interesting... it looks like that TypeScript enums are transpiled into specific JavaScript code, which increases the generated JavaScript code in size, which doesn't happen with string literals. The only drawback is the lack of support for auto-completion in the IDE, I guess.

Maybe we can even scrap all of that completely and just handle the unlock result as a string since it is not really "used" at the moment. 😄

Reply:
> The only drawback is the lack of support for auto-completion in the IDE, I guess.

VSCode has no problem autocompleting such strings, btw.

<img width="411" alt="Bildschirmfoto 2023-05-15 um 12 30 14" src="https://github.com/cryptomator/hub/assets/1204330/313a2f9c-c6dd-452b-a6b5-bcdf86177baa">

Reply:
Fixed in affd91798e270961e87804fb03a3e88cdc4f5613